### PR TITLE
Transform classic loops to range-based for loops in tools

### PR DESCRIPTION
**Pull request by SunBlack**
Changes are based on the result of run-clang-tidy -header-filter='.*' -checks='-*,modernize-loop-convert' -fix

### CODE REVIEW

File: tools/obj_rec_ransac_orr_octree.cpp [hunk 1]
```diff
@@ -157,10 +157,10 @@ void updateViewer (ORROctree& octree, PCLVisualizer& viz, std::vector<ORROctree:
   int i = 0;
 
   // Show the cubes
-  for ( std::list<ORROctree::Node*>::iterator it = intersected_leaves.begin () ; it != intersected_leaves.end () ; ++it )
+  for (const auto &intersected_leave : intersected_leaves)
```
Comment (SergioRAgostinho) [inline tools/obj_rec_ransac_orr_octree.cpp]: Should leaf and not leave.

File: tools/obj_rec_ransac_orr_octree.cpp [hunk 2]
```diff
@@ -157,10 +157,10 @@ void updateViewer (ORROctree& octree, PCLVisualizer& viz, std::vector<ORROctree:
...
   {
     sprintf(cube_id, "cube %i", ++i);
-    b = (*it)->getBounds ();
+    b = intersected_leave->getBounds ();
```
Comment (SergioRAgostinho) [inline tools/obj_rec_ransac_orr_octree.cpp]: Same here

File: tools/obj_rec_ransac_orr_octree.cpp [hunk 3]
```diff
@@ -320,9 +320,9 @@ void show_octree (ORROctree* octree, PCLVisualizer& viz, bool show_full_leaves_o
   if ( show_full_leaves_only )
   {
     std::vector<ORROctree::Node*>& full_leaves = octree->getFullLeaves ();
-    for ( std::vector<ORROctree::Node*>::iterator it = full_leaves.begin () ; it != full_leaves.end () ; ++it )
+    for (auto &full_leave : full_leaves)
```
Comment (SergioRAgostinho) [inline tools/obj_rec_ransac_orr_octree.cpp]: Same here.

File: tools/obj_rec_ransac_orr_octree.cpp [hunk 4]
```diff
@@ -320,9 +320,9 @@ void show_octree (ORROctree* octree, PCLVisualizer& viz, bool show_full_leaves_o
...
       // Add it to the other cubes
-      node_to_cube (*it, append);
+      node_to_cube (full_leave, append);
```
Comment (SergioRAgostinho) [inline tools/obj_rec_ransac_orr_octree.cpp]: Same here.

File: tools/obj_rec_ransac_orr_octree_zprojection.cpp [hunk 5]
```diff
@@ -195,9 +195,9 @@ void show_octree (ORROctree* octree, PCLVisualizer& viz)
   cout << "There are " << octree->getFullLeaves ().size () << " full leaves.\n";
 
   std::vector<ORROctree::Node*>& full_leaves = octree->getFullLeaves ();
-  for ( std::vector<ORROctree::Node*>::iterator it = full_leaves.begin () ; it != full_leaves.end () ; ++it )
+  for (auto &full_leave : full_leaves)
```
Comment (SergioRAgostinho) [inline tools/obj_rec_ransac_orr_octree_zprojection.cpp]: Same here.

File: tools/obj_rec_ransac_orr_octree_zprojection.cpp [hunk 6]
```diff
@@ -195,9 +195,9 @@ void show_octree (ORROctree* octree, PCLVisualizer& viz)
...
     // Add it to the other cubes
-    node_to_cube (*it, append);
+    node_to_cube (full_leave, append);
```
Comment (SergioRAgostinho) [inline tools/obj_rec_ransac_orr_octree_zprojection.cpp]: Same here.

File: tools/obj_rec_ransac_result.cpp [hunk 7]
```diff
@@ -241,8 +241,8 @@ update (CallbackParameters* params)
 {
   // Clear the visualizer from old object instances
   vtkRenderer *renderer = params->viz_.getRenderWindow ()->GetRenderers ()->GetFirstRenderer ();
-  for ( list<vtkActor*>::iterator it = params->actors_.begin () ; it != params->actors_.end () ; ++it )
-    renderer->RemoveActor (*it);
+  for (auto &actor : params->actors_)
```
Comment (SergioRAgostinho) [inline tools/obj_rec_ransac_result.cpp]: Can this be const? I think I just saw some code above in which it was.

File: tools/obj_rec_ransac_result.cpp [hunk 8]
```diff
@@ -151,7 +151,7 @@ run (float pair_width, float voxel_size, float max_coplanarity_angle)
   list<vtkSmartPointer<vtkPolyData> > vtk_models_list;
 
   // Load the models and add them to the recognizer
-  for ( list<string>::iterator it = model_names.begin () ; it != model_names.end () ; ++it )
+  for (auto &model_name : model_names)
```
Comment (taketwo) [inline tools/obj_rec_ransac_result.cpp]:
```suggestion
  for (const auto &model_name : model_names)
```

**Pull request merged**